core/src/rpc.rs

```diff
@@ -850,19 +850,28 @@ impl JsonRpcRequestProcessor {
             ));
         }
         let mut data = account.data.to_vec();
-        let balance = spl_token_v1_0::state::unpack(&mut data)
-            .map_err(|_| {
+        let token_account =
+            spl_token_v1_0::state::unpack::<TokenAccount>(&mut data).map_err(|_| {
                 Error::invalid_params("Invalid param: not a v1.0 Token account".to_string())
-            })
-            .map(|account: &mut TokenAccount| account.amount)?;
+            })?;
+        let mint_account = bank
```

Comment:
We need to check for `native_mint::id()` and then use `native_mint::DECIMALS` to avoid hitting "Invalid param: could not find mint"